README.md

```diff
@@ -14,12 +14,23 @@ If there is anything you are looking for that is not covered here you should ref
 
 ### File structure ###
 
-You should not define multiple public/internal types (ie class, struct, enum) in the same file; each type should have its own file.
+You should not define multiple public/internal types (ie class, struct, enum) in the same file; each type should have its own file. If an enum or protocol is related to the class but has to be known outside define it inside the class file.
```

Comment:
I think we discussed this and the idea is, given that enums are their own types, they should be in their file, regardless of whether they are a "part" of a class or not. I feel like this definition is too vague -- because even if the enum is only declared as a property in one area, that doesn't mean that other classes use it. It also then opens up issues of how to deal with refactoring enums into their own file when they are used -- I feel like this is will not happen and make this point moot.

Since types are first-class citizens in Swift, I believe they should always be in their own code file unless they are private (for obvious reasons).